packages/alfa-css/src/value/calculation.ts

```diff
@@ -0,0 +1,848 @@
+
+      if (isPercentage(value)) {
+        return Kind.of("percentage");
+      }
+
+      if (isLength(value)) {
+        return Kind.of("length");
+      }
+
+      if (isAngle(value)) {
+        return Kind.of("angle");
+      }
+
+      return Kind.of();
+    }
+
+    public get value(): Numeric {
+      return this._value;
+    }
+
+    public reduce(resolve: Mapper<Numeric>): Expression {
+      const value = this._value;
+
+      if (isLength(value) && value.isAbsolute()) {
+        return Value.of(value.withUnit("px"));
+      }
+
+      if (isAngle(value)) {
+        return Value.of(value.withUnit("deg"));
+      }
+
+      return Value.of(resolve(value));
+    }
+
+    public equals(value: unknown): value is this {
+      return value instanceof Value && value._value.equals(this._value);
+    }
+
+    public toJSON(): Value.JSON {
+      return {
+        type: "value",
+        value: this._value.toJSON(),
+      };
+    }
+
+    public toString(): string {
+      return `${this._value}`;
+    }
+  }
+
+  export namespace Value {
+    export interface JSON extends Expression.JSON {
+      type: "value";
+      value: Numeric.JSON;
+    }
+  }
+
+  export function isValueExpression(
+    expression: Expression
+  ): expression is Value {
+    return expression.type === "value";
+  }
+
+  /**
+   * @see https://drafts.csswg.org/css-values/#calculation-tree-operator-nodes
+   */
+  export abstract class Operation<
+    O extends Array<Expression> = Array<Expression>
+  > extends Expression {
+    protected readonly _operands: Readonly<O>;
+    protected readonly _kind: Kind;
+
+    protected constructor(operands: Readonly<O>, kind: Kind) {
+      super();
+
+      this._operands = operands;
+      this._kind = kind;
+    }
+
+    public get operands(): Readonly<O> {
+      return this._operands;
+    }
+
+    public get kind(): Kind {
+      return this._kind;
+    }
+
+    public equals(value: this): value is this {
+      return (
+        value instanceof Operation &&
+        value.type === this.type &&
+        value._operands.length === this._operands.length &&
+        value._operands.every((operand, i) => operand.equals(this._operands[i]))
+      );
+    }
+
+    public toJSON(): Operation.JSON {
+      return {
+        ...super.toJSON(),
+        operands: this._operands.map((operand) => operand.toJSON()),
+      };
+    }
+  }
+
+  export namespace Operation {
+    export interface JSON extends Expression.JSON {
+      operands: Array<Expression.JSON>;
+    }
+
+    export abstract class Unary extends Operation<[Expression]> {
+      protected constructor(operands: [Expression], kind: Kind) {
+        super(operands, kind);
+      }
+    }
+
+    export abstract class Binary extends Operation<[Expression, Expression]> {
+      protected constructor(operands: [Expression, Expression], kind: Kind) {
+        super(operands, kind);
+      }
+    }
+  }
+
+  export class Sum extends Operation.Binary {
+    public static of(
```

Comment:
Could be overloaded as `of(fst: Expression, snd: Expression)` for flexibility of use (same with other binary operations).

Reply:
That's equivalent to the current signature; `foo(...args: [string, number])` expands to `foo(args_0: string, args_1: number)`.